client_apis/python/example/sensor_isolation_alert.py

```diff
@@ -0,0 +1,176 @@
+#  the state of their network isolation since the last running of this script.
+# Run this script via a job scheduler, such as cron, to be notified 
+#  when a sensor's network isolation state has changed.
+# The script will track isolated sensors between runs via isolated_sensors.txt
+
+__author__ = 'BJSwope'
+import sys
+import optparse
+import warnings
+import smtplib
+import cbapi
+import json
+import collections
+import socket
+from email.mime.text import MIMEText
+
+def build_cli_parser():
+    parser = optparse.OptionParser(usage="%prog [options]", description="Dump sensor list")
+
+    # for each supported output type, add an option
+    #
+    parser.add_option("-c", "--cburl", action="store", default=None, dest="url",
+                      help="CB server's URL.  e.g., http://127.0.0.1 ")
+    parser.add_option("-a", "--apitoken", action="store", default=None, dest="token",
+                      help="API Token for Carbon Black server")
+    parser.add_option("-n", "--no-ssl-verify", action="store", default=False, dest="ssl_verify",
+                      help="Do not verify server SSL certificate.")
+    parser.add_option("-g", "--group", action="store", default=None, dest="groupid",
+                      help="Limit sensor listing to just those specified by the sensor group id provided")
+    return parser
+
+
+def send_mail(sensor):
+    mail = {}
+    if sensor['network_isolation_enabled'] == True: 
+        if sensor['is_isolating'] == True:
+            # Isolation Enabled and Active email
+            msg="Network Isolation enabled and active!\r\n Host: %s\r\nCarbon Black Console: %s\r\n Last Check-In Time: %s\r\n" \
+            % (sensor['computer_name'], sensor['url'], sensor['last_checkin_time'])
+            msg = MIMEText(msg)
+            msg['Subject'] = 'Host Isolation Activated By Carbon Black'
+        else:
+            # Isolation Enabled but Not Active email
+            msg="Network Isolation enabled and will activate at next sensor check in.\r\n Host: %s\r\nCarbon Black Console: %s\r\n Last Check-In Time: %s\r\nNext Check-In Time: %s" \
+            % (sensor['computer_name'], sensor['url'], sensor['last_checkin_time'], sensor['next_checkin_time'])
+            msg = MIMEText(msg)
+            msg['Subject'] = 'Host Isolation Enabled By Carbon Black'
+    elif sensor['network_isolation_enabled'] == False:
+        # Isolation Disabled email
+        msg="Network Isolation disabled and will deactivate at next sensor check in.\r\n Host: %s\r\nCarbon Black Console: %s\r\n Last Check-In Time: %s\r\nNext Check-In Time: %s" \
+            % (sensor['computer_name'], sensor['url'], sensor['last_checkin_time'], sensor['next_checkin_time'])
+        msg = MIMEText(msg)
+        msg['Subject'] = 'Host Isolation Disabled By Carbon Black'
+    else:
+        return
+    hostname = socket.getfqdn()
+    msg['From'] = 'sensor_isolation@%s' % (hostname)
+    msg['To'] = 'bj@carbonblack.com'
```

Comment:
You'll want to make this a configurable option :) Same with the SMTP server, and from address.